src/Microsoft.DotNet.Build.Tasks/PackageFiles/clean.targets

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <UserLocalFolder Condition="'$(OsEnvironment)'!='Unix'">$(LocalAppData)/</UserLocalFolder>
+    <UserLocalFolder Condition="'$(OsEnvironment)'=='Unix'">$(HOME)/.local/share/</UserLocalFolder>
+  </PropertyGroup>
+
+  <Target Name="CleanPackages">
+    <RemoveDir Directories="$(PackagesDir)" />
+  </Target>
+
+  <Target Name="CleanPackagesCache">
+    <RemoveDir Directories="$(UserLocalFolder)NuGet/Cache/;$(UserLocalFolder)NuGet/v3-cache/;$(UserLocalFolder)dnu/cache/" />
```

Comment:
I tested it and it simply no-ops if one of the directories doesn't exist.  This implementation is consistent with the definition of the Clean target in build.proj in CoreFX.